AsyncDisplayKit/ASRunLoopQueue.mm

```diff
@@ -102,6 +126,12 @@ - (void)processQueue
       self.queueConsumer(itemsToProcess[i], isQueueDrained);
     }
   }
+
+  // If the queue is not fully drained yet force another run loop to process next batch of items
+  if (!isQueueDrained) {
+    CFRunLoopSourceSignal(_runLoopSource);
+    CFRunLoopWakeUp(_runLoop);
```

Comment:
Do you need the WakeUp call?  I suspect the answer is no, because at this time we are currently running in an observer callback, and after running all of the BeforeWaiting observers it must surely check for more sources / etc to run before determining if it will sleep?  I could be wrong, but let's remove it if we don't think it is required as it will help keep the code as simple as possible.